mypy/test/testcheck.py

```diff
@@ -103,18 +102,16 @@ def run_case(self, testcase: DataDrivenTestCase) -> None:
                         raise ValueError(
                             'Output file {} exists though test case only has {} runs'.format(
                                 file, num_steps))
+            steps = testcase.find_steps()
             for step in range(1, num_steps + 1):
-                self.run_case_once(testcase, step)
+                ops = steps[step - 2] if step - 2 < len(steps) else []
```

Comment:
For step 1, the index will be -1, which seems wrong. Apparently the value of `ops` is ignored in that case so this doesn't affect behavior, but it's still worth fixing.